src/python/zquantum/optimizers/simple_gradient_descent.py

```diff
@@ -42,11 +42,20 @@ def _minimize(
         Finds the parameters which minimize given cost function, by trying all
         the parameters from the provided list of points.
 
+        Note:
+            This optimizer does not require evaluation of the cost function,
+            but relies only on gradient evaluation. This means, that if we want to
+            keep track of values of the cost functions for each iteration, we
+            need to perform extra evaluations. Therefore using `keep_history=True`
+            will increase add extra evaluations that are not necessary for
```

Comment:
I'm not sure if this sentence is correct. Shouldn't it be "Therefore using `keep_history=True` will ~increase~ add extra evaluations that are not necessary for the optimization process ~itslelf~ itself?

Reply:
Will agree with the first point, but I really liked the spelling of "itslelf" :P

Reply:
Fixed in https://github.com/zapatacomputing/z-quantum-optimizers/pull/79/commits/a88fde62b7cbbf261056b185e6375f1c9b372543